templates/chiffee/purchase.html

```diff
@@ -0,0 +1,94 @@
+{% extends "chiffee/base.html" %}
+
+{% load chiffee_tags %}
+
+{% block body %}
+    <form action="{% url 'chiffee:make-purchases' %}" method="post">
+        {% csrf_token %}
+
+        <p>Wie viele willst du kaufen?</p>
+
+        {% for i in quantity_range %}
+            {% if i == quantity %}
+                <button class="active btn btn-outline-info btn-lg"
+                        type="submit"
+                        name="edit"
+                        value="{{ i }}">
+                    {{ i }}
+                </button>
+            {% else %}
+                <button class="btn btn-outline-info btn-lg"
+                        type="submit"
+                        name="edit"
+                        value="{{ i }}">
+                    {{ i }}
+                </button>
+            {% endif %}
+        {% endfor %}
+
+        <br>
+        <br>
+
+        <p>Produkt: {{ product.name }}</p>
+        <p>Gesamtpreis: €{{ total_price|floatformat:-2 }}</p>
+
+        <br>
+    </form>
+
+    <form action="{% url 'chiffee:confirm-purchases' %}" method="post">
+        {% csrf_token %}
+
+        {% if not user.is_authenticated %}
+            {% for u in users %}
+                {% if forloop.counter0|divisibleby:4 %}
+                    <div class="row">
+                {% endif %}
+
+            <div class="col-3">
+                {% if u.is_active %}
+                    {% if u|has_group:'professors' %}
+                        <button class="btn btn-block btn-lg btn-menu btn-primary"
+                                name="confirm"
+                                type="submit"
+                                value="{{ u.username }}">
+                            {{ u.first_name }} {{ u.last_name }}
+                        </button>
+                    {% elif u|has_group:'employees' %}
+                        <button class="btn btn-block btn-lg btn-menu btn-success"
+                                name="confirm"
+                                type="submit"
+                                value="{{ u.username }}">
+                            {{ u.first_name }} {{ u.last_name }}
+                        </button>
+                    {% elif u|has_group:'students' %}
+                        <button class="btn btn-block btn-danger btn-lg btn-menu"
+                                name="confirm"
+                                type="submit"
+                                value="{{ u.username }}">
+                            {{ u.first_name }} {{ u.last_name }}
+                        </button>
+                    {% endif %}
```

Comment:
These three look very similar, what is the difference? And can it be done without copying?

Reply:
I'm not sure it's possible without JavaScript. The difference is the button class ('btn-danger' etc.), it colors them differently.

Reply:
what about:
```
<button class="btn btn-block btn-lg btn-menu {% if u|has_group:'professors' %}btn-primary{% elif u|has_group:'employees' %}btn-success{% elif u|has_group:'students' %}btn-menu{% endif %}"
                                name="confirm"
                                type="submit"
                                value="{{ u.username }}">
                            {{ u.first_name }} {{ u.last_name }}
                        </button>
```

just with nicer formatting.

Reply:
Ah yeah, you can put it in tags too. :D Will do in a bit.

Reply:
Fixed.